src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -120,6 +120,20 @@ public boolean containsEntryWithId(String id) {
         return internalIDs.contains(id);
     }
 
+    /**
+     * Returns whether an entry with the given bibtex key exists.
+     */
+    public synchronized boolean containsEntryWithKey(String key) {
+        int keyHash = key.hashCode(); // key hash for better performance
```

Comment:
After checking that code I decided to write this method. I think it will be
noticeable on large databases. (getEntryByName always goes through the
whole database and returns the last entry with that key).

I was thinking about the optional thing though. :-)